internal/plugins/darwin/processorinfo_test_arm64.go

```diff
@@ -0,0 +1,111 @@
+// Copyright 2022 New Relic Corporation. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+//go:build darwin
+// +build darwin
+
+package darwin
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/newrelic/infrastructure-agent/internal/agent"
+	testing2 "github.com/newrelic/infrastructure-agent/internal/plugins/testing"
+	"github.com/newrelic/infrastructure-agent/pkg/plugins/ids"
+	"github.com/newrelic/infrastructure-agent/pkg/sysinfo/cloud"
+	"gotest.tools/assert"
+)
+
+var ErrUnknownCommand = errors.New("unknown command")
+
+func TestData(t *testing.T) {
+	t.Parallel()
+	testCases := []struct {
+		name                 string
+		unameOutput          string
+		systemProfilerOutput string
+		expectedData         HostInfoData
+	}{
+		{
+			name:        "Arm arch output",
+			unameOutput: "Darwin Kernel Version 21.6.0: Wed Aug 10 14:28:23 PDT 2022; root:xnu-8020.141.5~2/RELEASE_ARM64_T6000",
+			systemProfilerOutput: `Hardware:
+
+    Hardware overview:
+
+      Model Name: MacBook Pro
+      Model Identifier: MacBookPro18,4
+      Chip: Apple M1 Max
+      Total Number of Cores: 10 (8 performance and 2 efficiency)
+      Memory: 64 GB
+      System Firmware Version: 7459.141.1
+      OS Loader Version: 7459.141.1
+      Serial Number (system): H9AAAAA9WK
+      Hardware UUID: E62094F3-9A33-5555-5555-F4C3A1E16AC5
```

Comment:
Just in case, I would fake this data (if it is harvested from one of our machines, same for intel test)